components/eamxx/src/physics/shoc/eamxx_shoc_process_interface.cpp

```diff
@@ -461,5 +465,53 @@ void SHOCMacrophysics::finalize_impl()
   // Do nothing
 }
 // =========================================================================================
+void SHOCMacrophysics::check_flux_state_consistency(const double dt)
+{
+  using PC = scream::physics::Constants<Real>;
+  const Real gravit = PC::gravit;
+  const Real qmin   = 1e-12; // minimum permitted constituent concentration (kg/kg)
+
+  const auto& pseudo_density = get_field_in ("pseudo_density").get_view<const Spack**>();
+  const auto& surf_evap      = get_field_out("surf_evap").get_view<Real*>();
+  const auto& qv             = get_field_out("qv").get_view<Spack**>();
 
+  const auto nlevs           = m_num_levs;
+  const auto nlev_packs      = ekat::npack<Spack>(nlevs);
+  const auto last_pack_idx   = (nlevs-1)/Spack::n;
+  const auto last_pack_entry = (nlevs-1)%Spack::n;
+  const auto policy          = ekat::ExeSpaceUtils<KT::ExeSpace>::get_default_team_policy(m_num_cols, nlev_packs);
+  Kokkos::parallel_for("check_flux_state_consistency",
+                       policy,
+                       KOKKOS_LAMBDA (const KT::MemberType& team) {
+    const auto i = team.league_rank();
+
+    const auto& pseudo_density_i = ekat::subview(pseudo_density, i);
+    const auto& qv_i             = ekat::subview(qv, i);
+
+    // reciprocal of pseudo_density at the bottom layer
+    const auto rpdel = 1.0/pseudo_density_i(last_pack_idx)[last_pack_entry];
+
+    // Check if the negative surface latent heat flux can exhaust
+    // the moisture in the lowest model level. If so, apply fixer.
+    const auto condition = surf_evap(i) - (qmin - qv_i(last_pack_idx)[last_pack_entry])/(dt*gravit*rpdel);
+    if (condition < 0) {
+      const auto cc = abs(surf_evap(i)*dt*gravit);
```

Comment:
For this case of a double, rather than pack, argument, I believe `abs` is C's `int abs(int)`, not C++'s `std::abs` call through to, essentially, `fabs`. Demo:
```C++
$ cat abs.cpp
#include <iostream>
int main () { std::cout << abs(-3.14) << "\n"; }
$ g++ -std=c++14 abs.cpp; if [ $? == 0 ]; then ./a.out; fi
3
$ cat abs.cpp
#include <iostream>
int main () { std::cout << std::abs(-3.14) << "\n"; }
$ g++ -std=c++14 abs.cpp; if [ $? == 0 ]; then ./a.out; fi
3.14
```